### ISSUE TITLE
Can not change scheduler to mq-deadline or mq-deadline-nodefault

### ISSUE THREAD

**Issue by BSerpent86** [damentz/liquorix-package#50]
Kernel: 
```
# uname -r
5.10.11-lqx1-1-lqx
```
on EndeavourOS/Arch

I can't change the scheduler to mq-deadline.
```
# echo mq-deadline > /sys/block/nvme0n1/queue/scheduler
bash: echo: write error: Invalid argument
```

First thing I noticed is, that it is called mq-deadline-nodefault:
```
# cat /sys/block/nvme0n1/queue/scheduler
mq-deadline-nodefault [bfq] none
```

but even when trying to change it to that, it won't work:
```
# echo mq-deadline-nodefault > /sys/block/nvme0n1/queue/scheduler
bash: echo: write error: Invalid argument
```

Changing to "none" however works:
```
# echo none > /sys/block/nvme0n1/queue/scheduler
# cat /sys/block/nvme0n1/queue/scheduler
[none] mq-deadline-nodefault bfq
```
Changing back to bfq also works:
```
# echo bfq > /sys/block/nvme0n1/queue/scheduler
# cat /sys/block/nvme0n1/queue/scheduler
mq-deadline-nodefault [bfq] none
```
Is there a specific reason why this kernel is doing that? 

dmesg:
```
dmesg | grep deadline
[    0.215813] io scheduler mq-deadline-nodefault registered
```

**Comment by damentz** [comment]
That is a good question, the intention of renaming mq-deadline was to prevent distribution stock udev rules and daemons from using mq-deadline against the preferences of the kernel.  It looks like renaming it isn't such a simple task.

With that in mind, XanMod solved this issue by making the module/subsystem that depended on mq-deadline depend on bfq instead.  If I make that change in Liquorix, mq-deadline won't be available whatsoever, so that might be the direction I go instead of figuring out what happened here.

**Comment by damentz** [comment]
One final thought, if you intend to get performance over responsiveness, just choose "none".  Not all consumer nvme drives can handle it well but it'll probably do what you want except in edge cases, like rsyncing at maximum speed while you're opening a browser after a cold boot.

**Comment by BSerpent86** [comment]
I can understand that, but I will have to stop using your kernel if that is your course of action. I need mq-deadline because of several reasons, one of them being that none and bfq give me interface lags and hangs on almost every task (I multitask a lot). Those things are not happening with mq-deadline.

**Comment by damentz** [comment]
@BSerpent86 is that a recent thing or has BFQ always been laggy?  BFQ is the only IO scheduler that guarantees responsiveness, so it's surprising that it's _worse_ than mq-deadline.

EDIT: I reverted a commit adopted from XanMod that may be causing the issue.  It was forcing multithreaded IO in all occasions, but that was probably causing BFQ to behave incorrectly in your use-cases.  Next release may resolve your issue.

**Comment by BSerpent86** [comment]
I always had that laggy issue on BFQ and none, it also happens on several SSD's, mainboards and CPU's from AMD and Intel.

The last release was worse though, thats why I checked schedulers and saw that it is not using mq-deadline anymore, although I always set it with an udev rule.

**Comment by damentz** [comment]
I see, must be some weird hardware interaction that's different with certain software/hardware combinations.

In the meantime, I think I spotted the issue and pushed up a fix to 5.10/master: https://github.com/zen-kernel/zen-kernel/commit/223b7e095efa96045c164f3bf3576e1d1f599946

A new kernel should come out in the next 2 hours that lets you change the IO scheduler to mq-deadline-nodefault.

**Comment by BSerpent86** [comment]
Thanks. The second update lets me select mq-deadline-nodefault again.
